examples/asr/asr_hybrid_transducer_ctc/helpers/convert_nemo_asr_hybrid_to_ctc.py

```diff
@@ -0,0 +1,170 @@
+        del new_cfg['labels']
+    elif (not BPE) and 'tokenizer' in new_cfg:
+        del new_cfg['tokenizer']
+    del new_cfg['target']
+    del new_cfg['nemo_version']
+
+    new_cfg_oc = OmegaConf.create(new_cfg)
+    ctc_model = ctc_class.restore_from(
+        args.input, map_location=torch.device('cpu'), override_config_path=new_cfg_oc, strict=False
+    )
+
+    assert all(
+        [
+            torch.allclose(hybrid_model.state_dict()[x], ctc_model.state_dict()[x])
+            for x in hybrid_model.state_dict().keys()
+            if x.split('.')[0] in ['preprocessor', 'encoder']
+        ]
+    ), "Encoder and preprocessor state dicts don't match!"
+
+    ctc_model.decoder.load_state_dict(hybrid_model.ctc_decoder.state_dict())
+
+    assert all(
+        [
+            torch.allclose(hybrid_model.ctc_decoder.state_dict()[x], ctc_model.decoder.state_dict()[x])
+            for x in hybrid_model.ctc_decoder.state_dict().keys()
+        ]
+    ), "Decoder state_dict load failed!"
+
+    assert isinstance(ctc_model, ctc_class), "Extracted CTC model is of the wrong expected class!"
+
+    return ctc_model
+
+
+def extract_model_rnnt(args, hybrid_model):
+    BPE = False
+    rnnt_class = EncDecRNNTModel
+    if 'tokenizer' in hybrid_model.cfg.keys():
+        BPE = True
+        rnnt_class = EncDecRNNTBPEModel
+
+    hybrid_model_cfg = OmegaConf.to_container(hybrid_model.cfg)
+
+    new_cfg = hybrid_model_cfg.copy()
+    del new_cfg['aux_ctc']
+    if BPE and 'labels' in new_cfg:
+        del new_cfg['labels']
+    elif (not BPE) and 'tokenizer' in new_cfg:
+        del new_cfg['tokenizer']
+    del new_cfg['target']
+    del new_cfg['nemo_version']
+
+    new_cfg_oc = OmegaConf.create(new_cfg)
+    rnnt_model = rnnt_class.restore_from(
+        args.input, map_location=torch.device('cpu'), override_config_path=new_cfg_oc, strict=False
+    )
+
+    assert all(
+        [
+            torch.allclose(hybrid_model.state_dict()[x], rnnt_model.state_dict()[x])
+            for x in hybrid_model.state_dict().keys()
+            if x.split('.')[0] in ['preprocessor', 'encoder', 'decoder', 'joint']
+        ]
+    ), "State dict values mismatch, something went wrong!"
+
+    assert isinstance(rnnt_model, rnnt_class), "Extracted RNNT model is of the wrong expected class!"
+
+    return rnnt_model
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-i', '--input', required=True, type=str, help='path to Nemo Hybrid model .nemo file')
+    parser.add_argument('-o', '--output', required=True, type=str, help='path and name of output .nemo file')
+    parser.add_argument(
+        '-m',
+        '--model',
+        required=False,
+        type=str,
+        default='ctc',
+        choices=['ctc', 'rnnt'],
+        help='whether to output a ctc or rnnt model from the hybrid',
+    )
+    parser.add_argument('--cuda', action='store_true', help='put Nemo model onto GPU prior to savedown')
```

Comment:
Does this do anything to the state dict ? All tensors would be marked as CPU when calling model.state_dict() no ?

Reply:
No, if you do model.cuda(), then all of the tensors in model.state_dict() do have their device as cuda. To be honest though, I'm not sure why anybody would ever need to use this, because you can control the map_location in restore_from anyway. I'm happy to remove it if you agree.

Reply:
Hmm lets remove

Reply:
Okay, I have removed it, can you please re-approve?